core/frontend/src/render/webgl/glsl/Clipping.ts

```diff
@@ -66,8 +69,41 @@ const applyClipPlanesPostlude = `
     } else if (!clippedByCurrentPlaneSet && calcClipPlaneDist(v_eyeSpace, plane) < 0.0) {
       clippedByCurrentPlaneSet = true;
     }
+
+    //avoiding near and far clip planes
+    if ((i <= u_clipParams[1] - 2) && (!clippedByCurrentPlaneSet)) {
```

Comment:
This all works as intended now, the one thing we might want to change is adding some sort of condition to avoid doing these calculations on the majority of frags. Something like:

if calcClipPlaneDist(v_eyeSpace, plane) >= someArbitraryBarrier    -> don't do calculations